app/models/post.rb

```diff
@@ -0,0 +1,19 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  def update_user_posts_counter
+    author.update(post_count: author.authored_posts.count)
+  end
+
```

Comment:
- [ ] The purpose of these counter functions is to increase the counter every time a post is saved in the database. Kindly make use of the model callback functions such as `after_save`. This function will save a counter every time a post is saved.

![counter](https://github.com/danielochuba/duo-blog/assets/53124548/44c981e9-ee4e-4394-99bd-541b839e3a7a)


```ruby
  after_save :update_user_posts_counter
  private

  def update_user_posts_counter
    author.increment!(:post_counter)
  end
```